src/components/Timetable/UploadPage.jsx

```diff
@@ -35,7 +37,7 @@ export default function UploadPage() {
         InputProps={{
           endAdornment: (
             <InputAdornment position="end">
-              <IconButton edge="end" color="primary" aria-label="submit">
+              <IconButton edge="end" color="primary" aria-label="submit" onClick={sendTimetableURL}>
```

Comment:
```suggestion
              <IconButton edge="end" color="primary" aria-label="submit" onClick={() => sendTimetableURL(calendarURL)}>
```

Reply:
@arathibiju 
would be nice if you can make it a method
```js
const handleSubmit = () => {
   sendTimetableURL(calendarURL)}
}
```

Reply:
Also would be nice to add tests for submitting

Reply:
Hiya, I've added some changes as requested as well as a sample test. Let me know if this is alright or if more changes are needed @scorpionknifes